vita3k/gui/src/controllers_dialog.cpp

```diff
@@ -173,6 +173,39 @@ static void add_bind_to_table(GuiState &gui, EmuEnvState &emuenv, const SDL_Game
     ImGui::PopID();
 }
 
+static void swap_controller_ports(CtrlState &state, int source_port, int dest_port) {
+    // Check if the ports are valid
+    if (source_port < 1 || source_port > SCE_CTRL_MAX_WIRELESS_NUM || dest_port < 1 || dest_port > SCE_CTRL_MAX_WIRELESS_NUM) {
+        LOG_ERROR("Ports are not valid.");
+        return;
+    }
+
+    // Find the controllers corresponding to the source and destination ports
+    auto source_controller_it = std::find_if(state.controllers.begin(), state.controllers.end(),
+        [source_port](const auto &pair) { return pair.second.port == source_port; });
+    auto dest_controller_it = std::find_if(state.controllers.begin(), state.controllers.end(),
+        [dest_port](const auto &pair) { return pair.second.port == dest_port; });
+
+    // Check that both controllers exist
+    if (source_controller_it == state.controllers.end() || dest_controller_it == state.controllers.end()) {
+        LOG_ERROR("Unable to find one or more controllers on the specified ports.");
+        return;
+    }
+
+    LOG_INFO("Controller on source port {} {} swapped with controller on destination port {} {}", source_port, state.controllers_name[source_port - 1], dest_port, state.controllers_name[dest_port - 1]);
+
+    // Swap controllers info and player index
+    SDL_GameControllerSetPlayerIndex(source_controller_it->second.controller.get(), dest_port - 1);
+    std::swap(source_controller_it->second.port, dest_controller_it->second.port);
+    std::swap(source_controller_it->second.has_accel, dest_controller_it->second.has_accel);
+    std::swap(source_controller_it->second.has_gyro, dest_controller_it->second.has_gyro);
+    std::swap(source_controller_it->second.has_led, dest_controller_it->second.has_led);
+
+    // Swap controller names
+    std::swap(state.controllers_name[source_port - 1], state.controllers_name[dest_port - 1]);
+    std::swap(state.controllers_has_motion_support[source_port - 1], state.controllers_has_motion_support[dest_port - 1]);
+}
+
```

Comment:
There are like 10 different issues with this code. Can't you just swap the port value and be done?

Reply:
Hmmm 🤔 i check

Reply:
Done, sorry, i have not see this error, for’other is obligatory for controller name keep same
If now switch name it keep dualsense when is ds4 after swap